iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -1085,20 +1241,25 @@ protected override void Dispose(bool disposing)
 
                 if (!_isDisposed)
                 {
+                    _isDisposing = true;
+
                     base.Dispose(disposing);
                     if (disposing)
                     {
                         _handleDisconnectCts?.Cancel();
                         _handleDisconnectCts?.Dispose();
-                        if (_handlerSemaphore != null && _handlerSemaphore.CurrentCount == 0)
-                        {
-                            _handlerSemaphore.Release();
-                        }
-                        _handlerSemaphore?.Dispose();
-                        _handlerSemaphore = null;
+
+                        _cancelPendingOperations?.Cancel();
```

Comment:
Is there any chance one of these could throw, and we'd fail to dispose the remaining set?

Reply:
I can wrap these calls within try-catch block